iceoryx_examples/singleprocess/single_process.cpp

```diff
@@ -44,39 +45,57 @@ void consoleOutput(const std::string& output)
 
 void sender()
 {
-    iox::popo::Publisher publisher({"Single", "Process", "Demo"});
+    iox::popo::UntypedPublisher publisher({"Single", "Process", "Demo"});
     publisher.offer();
 
     uint64_t counter{0};
     while (keepRunning.load())
     {
-        auto sample = static_cast<TransmissionData_t*>(publisher.allocateChunk(sizeof(TransmissionData_t)));
-        sample->counter = counter++;
-        consoleOutput(std::string("Sending: " + std::to_string(sample->counter)));
-        publisher.sendChunk(sample);
+        publisher.loan(sizeof(TransmissionData_t)).and_then([&](iox::popo::Sample<void>& sample) {
+            auto rawSample = static_cast<TransmissionData_t*>(sample.get());
```

Comment:
Strictly speaking undefined behavior: you have some random pieces of memory and assign them to a `TransmissionData_t` pointer. After that - without calling the constructor - you assign values to some members.
I would suggest here a placement new:
```cpp
new (rawSample) TransmissionData_t{counter++};
```

Reply:
@elfenpiff basically adding the placement new in between, right?!
```
auto rawSample = static_cast<TransmissionData_t*>(sample.get());
new (rawSample) TransmissionData_t{counter++};
rawSample->counter = counter++;
```

Reply:
@marthtz, even easier `auto rawSample = new (sample.get()) TransmissionData_t()`

Reply:
Using a typed publisher now.